Sources/Basic/SortedArray.swift

```diff
@@ -26,22 +26,43 @@ public struct SortedArray<Element>: CustomStringConvertible {
     
     /// Insert the given element, maintaining the sort order.
     public mutating func insert(_ newElement: Element) {
-        elements.append(newElement)
-        // FIXME: It is way too costly to sort again for just one element.
-        // We can use binary search to find the index for this element.
-        elements.sort(by: areInIncreasingOrder)
+        elements.insert(newElement, at: self.insertionIndex(for: newElement))
+    }
+    
+    fileprivate func insertionIndex(for element: Element) -> Index {
```

Comment:
This is also a good place to document that this is just doing a binary search for finding the index.